### PR TITLE
`TextDocument` archetype

### PR DESCRIPTION
**Pull request by emilk**
* Part of https://github.com/rerun-io/rerun/issues/2793

### What
* Rename `datatypes.Label` to `datatypes.utf8`
* Rename `components.Label` to `components.Text`
* Add new `archetypes.TextDocument { body: components.Text }`
 
image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3173) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/3173)
- [Docs preview](https://rerun.io/preview/5e785feff2c57c3f8b7080e55e59fef3cd98a009/docs) 
- [Examples preview](https://rerun.io/preview/5e785feff2c57c3f8b7080e55e59fef3cd98a009/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

**Comment by teh-cmc** [comment]
I'll handle the rebase on top of the new roundtrip stuff, force push incoming :warning:

File: crates/re_sdk/src/msg_sender.rs
Comment (teh-cmc) [inline crates/re_sdk/src/msg_sender.rs:1]: how did this guy come back to life :scream:

File: crates/re_space_view_text_document/src/view_part_system.rs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    }
+}
+
+impl ViewPartSystem for TextDocumentSystem {
+    fn archetype(&self) -> ArchetypeDefinition {
+        vec1::vec1![
+            re_types::archetypes::TextDocument::indicator_component(),
+            re_types::components::Text::name(),
+        ]
+    }
```
Comment (teh-cmc) [inline crates/re_space_view_text_document/src/view_part_system.rs:34]:
```suggestion
    fn archetype(&self) -> ArchetypeDefinition {
        // TODO(#3159): use actual archetype definition
        // TextDocument::all_components().try_into().unwrap()
        vec1::vec1![
            re_types::archetypes::TextDocument::indicator_component(),
            re_types::components::Text::name(),
        ]
    }

File: crates/re_space_view_text_document/src/view_part_system.rs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            )?;
+
+            // TODO(emilk): use `iter_required_component` instead, once it doesn't require Default
+            for text_entry in arch_view
+                .iter_optional_component::<re_types::components::Text>()?
+                .flatten()
+            {
+                let re_types::components::Text(text) = text_entry;
+                self.text_entries.push(TextDocumentEntry { body: text });
+            }
```
Comment (teh-cmc) [inline crates/re_space_view_text_document/src/view_part_system.rs:64]: This _should_ use `iter_required_component`, and `Text` should implement `Default`? At least until we integrate codegen with `MaybeInit` and stuff, which won't happen for a very long time.

File: docs/code-examples/text_document.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env python3
+"""Log a simple set of line segments."""
+
+import rerun as rr
+import rerun.experimental as rr2
+
+rr.init("rerun_example_text_document", spawn=True)
+
+rr2.log("text_document", rr2.TextDocument(
+    body="Hello, TextDocument!",
```
Comment (teh-cmc) [inline docs/code-examples/text_document.py]: remove the comma so `black` formats it all on a single line
```suggestion
    body="Hello, TextDocument!"

File: docs/code-examples/text_document.py [hunk 4]
```diff
@@ -6,6 +6,9 @@
...
 
-rr2.log("text_document", rr2.TextDocument(
-    body="Hello, TextDocument!",
-))
+rr2.log(
+    "text_document",
+    rr2.TextDocument(
+        body="Hello, TextDocument!",
+    ),
+)
```
Comment (teh-cmc) [inline docs/code-examples/text_document.py:14]: remove the commas so `black` formats it all as one line; this is pretty jarring as-is and will take a bunch of unneeded space in the docstring
```suggestion
rr2.log(
    "text_document",
    rr2.TextDocument(
        body="Hello, TextDocument!"
    )
)

File: tests/python/roundtrips/text_document/main.py [hunk 5]
```diff
@@ -17,9 +17,12 @@ def main() -> None:
...
 
-    rr2.log("text_document", rr2.TextDocument(
-        body="Hello, TextDocument!",
-    ))
+    rr2.log(
+        "text_document",
+        rr2.TextDocument(
+            body="Hello, TextDocument!",
+        ),
+    )
```
Comment (teh-cmc) [inline tests/python/roundtrips/text_document/main.py:25]: same here

**Review by teh-cmc — APPROVED** [review]
:+1: LGTM except for a required component being queried as optional.

File: docs/code-examples/text_document.py (hunk 4 above)
Comment (emilk) [reply docs/code-examples/text_document.py:14]: I had it all on one line until black formatted it like this for me

File: crates/re_space_view_text_document/src/view_part_system.rs (hunk 2 above)
Comment (emilk) [reply crates/re_space_view_text_document/src/view_part_system.rs:66]: Why does it need to implement `Default` though? I don't want to get a bunch of empty strings here!

If it is a technical limitation, then is there an issue for it?

Comment (emilk) [reply crates/re_space_view_text_document/src/view_part_system.rs:66]: Actually, I could just remove the `+ Default` bound and it works!

Comment (teh-cmc) [reply crates/re_space_view_text_document/src/view_part_system.rs]: Yes I was about to say that!

The issue is #2875 but we should already have fixed the issue for the deserialization path at this point, so you shouldn't even need the `Default` clause anymore.

**Pull request merged**